test/extended/util/client.go

```diff
@@ -356,13 +356,25 @@ func (c *CLI) setupProject() string {
 		"builder",
 	}
 	defaultRoleBindings := []string{"system:image-pullers", "system:image-builders"}
-	dcEnabled, err := IsCapabilityEnabled(c, configv1.ClusterVersionCapabilityDeploymentConfig)
+
+	// If image registry is enabled, continue checking for DeploymentConfig capability otherwise
+	// set default service account and default role bindings to empty slice,
+	// the SA will not contain the docker secret and the role binding is not expected to be present.
+	imageRegistryEnabled, err := IsCapabilityEnabled(c, configv1.ClusterVersionCapabilityImageRegistry)
 	o.Expect(err).NotTo(o.HaveOccurred())
-	if dcEnabled {
-		framework.Logf("%v capability is enabled, adding 'deployer' SA to the list of default SAs", configv1.ClusterVersionCapabilityDeploymentConfig)
-		DefaultServiceAccounts = append(DefaultServiceAccounts, "deployer")
-		defaultRoleBindings = append(defaultRoleBindings, "system:deployers")
+	if imageRegistryEnabled {
+		dcEnabled, err := IsCapabilityEnabled(c, configv1.ClusterVersionCapabilityDeploymentConfig)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		if dcEnabled {
+			framework.Logf("%v capability is enabled, adding 'deployer' SA to the list of default SAs", configv1.ClusterVersionCapabilityDeploymentConfig)
+			DefaultServiceAccounts = append(DefaultServiceAccounts, "deployer")
+			defaultRoleBindings = append(defaultRoleBindings, "system:deployers")
+		}
+	} else {
+		DefaultServiceAccounts = []string{}
+		defaultRoleBindings = []string{}
 	}
+
 	for _, sa := range DefaultServiceAccounts {
 		framework.Logf("Waiting for ServiceAccount %q to be provisioned...", sa)
 		err = WaitForServiceAccountWithSecret(c.KubeClient().CoreV1().ServiceAccounts(newNamespace), sa)
```

Comment:
seems better to update `WaitForServiceAccountWithSecret` to handle the case when no image pull secrets will be created.

Reply:
That's a great point, updating it. However, I think we still gotta do the check here since there a role binding check after this that will also not work when image registry isn't present. At least from what I could find, that should only be populated when the image-registry operator is enabled via this [controller code](https://github.com/openshift/openshift-controller-manager/blob/master/pkg/cmd/controller/authorization.go#L41) that looks like it's [conditional based on this](https://github.com/openshift/openshift-controller-manager/blob/master/pkg/cmd/openshift-controller-manager/controller_manager.go#L141)